lib/internal/policy/manifest.js

```diff
@@ -158,36 +167,66 @@ class Manifest {
         dependencyMap = ObjectCreate(null);
       }
       if (typeof dependencyMap === 'object' && !ArrayIsArray(dependencyMap)) {
+        // This is used so we don't traverse this every time
+        // in theory we can delete parts of the dep map once this is populated
+        const localMappings = new SafeMap();
         /**
-         * @returns {true | URL}
+         * @returns {true | null | URL}
          */
-        const dependencyRedirectList = (toSpecifier) => {
-          if (toSpecifier in dependencyMap !== true) {
+        const dependencyRedirectList = (specifier, conditions) => {
+          if (localMappings.has(specifier)) {
+            return localMappings.get(specifier);
+          }
+          if (specifier in dependencyMap !== true) {
+            localMappings.set(specifier, null);
             return null;
           }
-          const to = dependencyMap[toSpecifier];
-          if (to === true) {
+          let target = dependencyMap[specifier];
+          while (
+            target &&
+            typeof target === 'object' &&
+            !ArrayIsArray(target)
+          ) {
+            for (const candidateKey of ObjectKeys(target)) {
+              if (conditions.has(candidateKey)) {
+                target = target[candidateKey];
+                continue;
```

Comment:
I assume this `continue` is supposed to affect the `while` loop but I think it's continuing the for loop instead. It will continue to look up keys of the first target on subsequent target values.

Reply:
ah, yea, thats not good. can fix

Reply:
fixed in c9021ac15a